vector/src/Data/Vector.hs

```diff
@@ -2157,3 +2175,6 @@ unsafeCopy = G.unsafeCopy
 copy :: PrimMonad m => MVector (PrimState m) a -> Vector a -> m ()
 {-# INLINE copy #-}
 copy = G.copy
+
+-- $setup
+-- >>> default (Int)
```

Comment:
The problem is that `Integer` doesn't work with `Primitive`/`Unboxed`/`Storable`. Before, the default type was `Double` (except where `Int` was explicitly annotated), which I think is problematic, since floating point arithmetic isn't exact (for example 0.1 + 0.1 + 0.1 isn't exactly 0.3) and imo the `.0` is distracting/unnecessary (the numbers are all integers anyway).

Reply:
> The problem is that Integer doesn't work with Primitive/Unboxed/Storable.

I am suggesting this only for `Data.Vector` and `Data.Vector.Generic` (which uses `Data.Vector` in doctests)